pkg/kubelet/volume/populator/desired_state_of_world_populator.go

```diff
@@ -120,6 +124,32 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
 	for _, volumeToMount := range dswp.desiredStateOfWorld.GetVolumesToMount() {
 		if _, podExists :=
 			dswp.podManager.GetPodByUID(volumeToMount.Pod.UID); !podExists {
+			// Once a pod has been deleted from kubelet pod manager, do not
+			// delete it from volume manager. Instead call out to the kubelet
+			// containerRuntime interface to verify all containers in the pod
+			// have been terminated.
+
+			podStatus, podStatusErr := dswp.kubeContainerRuntime.GetPodStatus(
+				volumeToMount.Pod.UID,
+				volumeToMount.Pod.Name,
+				volumeToMount.Pod.Namespace)
+			if podStatusErr != nil {
+				glog.Errorf(
+					"kubeContainerRuntime.GetPodStatus returned error %v. Assuming pod %q is deleted.",
+					podStatusErr,
+					volumeToMount.PodName)
+			} else {
+				for _, containerStatus := range podStatus.ContainerStatuses {
+					if containerStatus.State != kubecontainer.ContainerStateExited {
+						glog.V(5).Infof(
+							"Pod %q has been removed from pod manager. However, it still has one or more containers (%q) in the non-exited state. Therefore it will not be removed from volume manager.",
+							volumeToMount.PodName,
+							containerStatus.ID)
+						continue
```

Comment:
Uh..... is this a `continue` for the outer `for` loop or the inner `for` loop? If it's only for the inner `for` loop, the code will continue to check the next status. I think this is incorrect.